app/controllers/concerns/commented.rb

```diff
@@ -0,0 +1,17 @@
+module Commented
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_comments, only: :show
```

Comment:
I'm worried about L12. It seems that it'd be safer to let CanCan load the resource and hook after that? Although it seems we're not using cancan for load_and_authorize

Maybe the concern is more trouble than it's worth?

L12 makes the assumption that `:find_or_initialize_X` is available, which will break in Pro's CardsController.

Alternatively maybe the concern can skip the filter and just provide a `def comments` method that's also a `helper_method` available to views. That way when it's called in the views, we're reasonably certain that the `instance_variable_get("@#{resource}")` is going to work, w/o making any assumptions about how it was found or initialised.

Reply:
I like (and used) the concern that just adds helper methods idea.